src/anndata/io/__init__.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings(
+        "ignore", message=r"Importing read_.* from `anndata` is deprecated"
+    )
```

Comment:
This seems a bit backwards. I’m OK with this approach, but I’m curious:

Is there anything that prevented doing this?

`io/__init__.py`
```py
from ._io.foo import read_x
from ._io.bar import read_y
...

__all__ = ["read_x", "read_y", ...]
```

`_io.__init__.py`
```py
def __getattr__(name: str) -> Any:
    from . import io
    attr = getattr(io, name)
    warnings.warn(...)
    return attr

__all__: list[str] = []
```

Reply:
Hmmm, hadn't thought of that.  I think that approach should be a bit cleaner...let me take a whack at it.  I'll re-request review if I land on something...

Reply:
Done, seems to work.